homeassistant/components/zha/__init__.py

```diff
@@ -369,6 +419,122 @@ def zdo_command(self, tsn, command_id, args):
         pass
 
 
+class ZHAEvent(object):
+    """When you want signals instead of entities.
+    Stateless sensors such as remotes are expected to generate an event
+    instead of a sensor entity in hass.
+    """
+
+    def __init__(self, hass, cluster, domain, discovery_info):
+        """Register callback that will be used for signals."""
+        self._hass = hass
+        self._cluster = cluster
+        self._cluster.add_listener(self)
+        ieee = discovery_info['endpoint'].device.ieee
+        ieeetail = ''.join(['%02x' % (o, ) for o in ieee[-4:]])
+        if discovery_info['manufacturer'] and discovery_info['model'] is not \
+                None:
+            self._id = "{}.{}_{}_{}_{}{}".format(
+                domain,
+                slugify(discovery_info['manufacturer']),
+                slugify(discovery_info['model']),
+                ieeetail,
+                discovery_info['endpoint'].endpoint_id,
+                discovery_info.get('entity_suffix', '')
+            )
+        else:
+            self._id = "{}.zha_{}_{}{}".format(
+                domain,
+                ieeetail,
+                discovery_info['endpoint'].endpoint_id,
+                discovery_info.get('entity_suffix', '')
+            )
+
+
+class ZHASwitchEvent(ZHAEvent):
+    """Switch / remote event for zha"""
+
+    def __init__(self, hass, cluster, discovery_info):
+        """Initialize Switch."""
+        super().__init__(hass, cluster, SWITCH, discovery_info)
+
+    @callback
+    def cluster_command(self, tsn, command_id, args):
+        """Handle commands received to this cluster."""
+        if command_id in (0x0000, 0x0040):
+            self._hass.bus.fire(
+                OFF_EVENT_KEY,
+                {DEVICE: self._id},
+                EventOrigin.remote
```

Comment:
What about either having an attribute indicating how many times the key was pressed in the last second or two (clicks_per_second), or maybe a separate event indicating double/triple clicks? eg for the 1st "off" command received in the last 2s time window we fire `OFF_EVENT_KEY`, for the 2nd command received in the last 2s windows, we fire  `OFF_DOUBLE_EVEN_KEY` or anything like this?
I could get some fancy automation with events like this.

Reply:
Maybe in a subsequent PR. For now I just want to focus on the functionality that was already here.